### PR TITLE
BUG: Repetition time had no effect on signal cleaning

### PR DESCRIPTION
**Pull request by KamalakerDadi**
This PR is continuation of #1629

### CODE REVIEW

File: nilearn/tests/test_signal.py [hunk 1]
```diff
@@ -274,6 +287,36 @@ def test_clean_detrending():
...
 
+def test_clean_t_r():
+    """Different TRs must produce different results after filtering"""
+    n_samples = 34
+    # n_features  Must be higher than 500
+    n_features = 501
+    x_orig = generate_signals_plus_trends(n_features=n_features,
+                                          n_samples=n_samples)
+    random_tr_list1 = np.round(np.random.rand(5) * 10, decimals=2)
+    random_tr_list2 = np.round(np.random.rand(5) * 10, decimals=2)
```
Comment (GaelVaroquaux) [inline nilearn/tests/test_signal.py]: There is an unseeded random number generator here. We cannot do that: every random number generator in the tests must be seeded.

Comment (GaelVaroquaux) [reply nilearn/tests/test_signal.py]: Also, that list is quite long: given that we test with two values for low and high pass filters below, do we need to draw 5 values? Maybe 3 is enough?

Comment (KamalakerDadi) [reply nilearn/tests/test_signal.py]: Thanks.. Good points. I pushed a change. Please have a look.

**Comment by GaelVaroquaux** [comment]
OK, this is great. Travis and Appveyor have passed.

I just released: we would need an entry in the changelog, with the bugfix and the credit to you and @raamana

**Comment by GaelVaroquaux** [comment]
Typo above: released => realized

**Comment by GaelVaroquaux** [comment]
Merging this guy: the changelog can be updated after.

**Comment by KamalakerDadi** [comment]
>I just released: we would need an entry in the changelog, with the bugfix and the credit to you and @raamana

Yup should be done with #1658

**Pull request merged**